src/app/_components/BlockList/LayoutA/useBlockList.ts

```diff
@@ -61,144 +62,93 @@ const createUIBlockList = (
-    useSuspenseBlocksByBurnBlock(secondToLastBurnBlock.burn_block_height, STX_BLOCK_LENGTH),
-    STX_BLOCK_LENGTH
-  );
+export function useBlockList(length: number) {
+  const queryClient = useQueryClient();
+  const { setFadeEffect, liveUpdates } = useBlockListContext();
 
-  return {
+  const {
     lastBurnBlock,
     secondToLastBurnBlock,
     lastBurnBlockStxBlocks,
     secondToLastBlockStxBlocks,
-  };
-}
+  } = useInitialBlockList();
+
+  const initialBlockHashes = useMemo(
+    () =>
+      new Set([
+        ...lastBurnBlockStxBlocks.map(block => block.hash),
+        ...secondToLastBlockStxBlocks.map(block => block.hash),
+      ]),
+    [lastBurnBlockStxBlocks, secondToLastBlockStxBlocks]
+  );
 
-export function useBlockList(
-  length: number,
-  liveUpdates: boolean,
-  setFadeEffect: (value: boolean) => void
-) {
-  const queryClient = useQueryClient();
-  const [displayedBlocks, setDisplayedBlocks] = React.useState<Record<string, number>>({});
-  const { latestBlocks, latestBlock, clearLatestBlocks } = useBlockListWebSocket(
-    liveUpdates,
-    setFadeEffect
+  const initialBurnBlockHashes = useMemo(
+    () => new Set([lastBurnBlock.burn_block_hash, secondToLastBurnBlock.burn_block_hash]),
+    [lastBurnBlock, secondToLastBurnBlock]
   );
 
-  const refetch = useCallback(
-    function () {
-      clearLatestBlocks();
-      return Promise.all([
+  const { latestBlock, latestBlocksCount, clearLatestBlocks } = useBlockListWebSocket(
+    initialBlockHashes,
+    initialBurnBlockHashes
+  );
+
+  const updateList = useCallback(
+    async function () {
+      setFadeEffect(true);
+      await Promise.all([
         queryClient.invalidateQueries({ queryKey: ['getBlocksByBurnBlock'] }),
         queryClient.invalidateQueries({ queryKey: ['burnBlocks'] }),
       ]);
+      clearLatestBlocks();
+      setFadeEffect(false);
     },
-    [queryClient]
+    [clearLatestBlocks, queryClient, setFadeEffect]
   );
 
+  const prevLiveUpdatesRef = useRef(liveUpdates);
+  const prevLatestBlocksCountRef = useRef(latestBlocksCount);
+
   useEffect(() => {
-    void refetch();
-  }, [liveUpdates, refetch]);
+    const liveUpdatesToggled = prevLiveUpdatesRef.current !== liveUpdates;
 
-  const {
-    lastBurnBlock,
-    secondToLastBurnBlock,
-    lastBurnBlockStxBlocks,
-    secondToLastBlockStxBlocks,
-  } = useBlockListData();
+    const receivedLatestBlockWhileLiveUpdates =
+      liveUpdates &&
+      latestBlocksCount > 0 &&
+      prevLatestBlocksCountRef.current !== latestBlocksCount;
 
-  if (liveUpdates) {
-    if (latestBlock && !displayedBlocks[latestBlock.hash]) {
+    if (liveUpdatesToggled) {
+      setFadeEffect(true);
+      clearLatestBlocks();
+      updateList().then(() => {
+        setFadeEffect(false);
+      });
+    } else if (receivedLatestBlockWhileLiveUpdates && latestBlock) {
       if (latestBlock.burn_block_height === lastBurnBlock.burn_block_height) {
```

Comment:
This seems like this is handling a blockchain specific issue, like a reorg. If so, it would be helpful to add a comment about that